Add tests in tests.py

SvetlaShakh opened the pull request:
Добавлены тесты в tests.py

Commit:
Created tests

SvetlaShakh committed:
Delete .coverage

SvetlaShakh committed:
Delete .idea directory

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,37 @@
+def collector_lst():
+    collector_lst= BooksCollector()
+
+    collector_lst.add_new_book('Гордость и предубеждение и зомби')
```
А столько книг нам нужно?

SvetlaShakh replied:
На каждый жанр добавляю две книги, что бы при сортировке он отбирал все книги с жанром, а не только первую  из списка с запрошенным жанром

AlexKitIt replied:
Прям овер, тут минимальный функционал, когда значений будет сотня все будем брать по 2?

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,37 @@
+
+@pytest.fixture
+def collector_lst():
+    collector_lst= BooksCollector()
+
+    collector_lst.add_new_book('Гордость и предубеждение и зомби')
```
Нужно исправить: это уже описано в фикстуре collector, необходимо переиспользовать а не копипастить

SvetlaShakh replied:
Исправила метод, добавила фикстуру collector

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,163 @@
+
+    def test_set_book_genre_not_from_books_genre_is_none(self, collector):
+
+        assert collector.set_book_genre('Henry Ford', 'Детективы') is None  # сообщение об ошибке/ ожидаемый результат
```
Нужно исправить: метод ничего не возвращает, его нельзя использовать в assert

SvetlaShakh replied:
заменила is на ==

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,163 @@
+
+        books_for_genre = []
+        for name, book_genre in collector_lst.books_genre.items():
+            if book_genre == genre:
```
Нужно исправить: здесь и далее: условий в тестах быть не может, тест всегда однозначен

SvetlaShakh replied:
Ожидаемый результат добавлен в список параметров. В остальных тестах словарь для проверки добавлен hardcode словарем (была функция которая его собирала)

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,163 @@
+# get_books_genre
+    def test_get_books_genre_get_dict_all_books(self, collector_lst):
+
+        assert collector_lst.get_books_genre() == collector_lst.books_genre
```
Можно лучше: для достоверности проверять стоит по хардкоженному словарю

SvetlaShakh replied:
Исправлено

Commit:
conftest.py :
 add fixture collector in fixture collector_lst
tests.py :
 test_set_book_genre_not_from_books_genre_is_none - change assert
 test_get_books_with_specific_genre_get_lst_books_for_genre - add expected resul in parametrize
 test_get_books_genre_get_dict_all_books - expected resul in hardcod list
 test_get_books_for_children_get_list_books_without_age_rating - expected resul in hardcod list

Commit:
Reformat code

Commit:
add gitignore file
remove .coverage file

SvetlaShakh committed:
Delete .idea directory

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,151 @@
+        assert collector.books_genre['Гордость и предубеждение и зомби'] == ''
+
+    def test_set_book_genre_not_from_books_genre_is_none(self, collector):
+        assert collector.set_book_genre('Henry Ford', 'Детективы') == None  # сообщение об ошибке/ ожидаемый результат
```
Нужно исправить: повторный коммент: 
метод set_book_genre выглядит так:
  def set_book_genre(self, name, genre):
        if name in self.books_genre and genre in self.genre:
            self.books_genre[name] = genre
            
 Он ничего не возвращает вообще, он просто добавляет жанр если книга есть, если нет - ничего не делает, нельзя его использовать в assert, тк в там идет сравнение результата выполнения метода с true, но как написано выше - это метод ничего не возвращает, нет разницы это is или ==

Commit:
test_set_book_genre_not_from_books_genre_is_none change expected result
collector_lst change number of books

Commit:
test_set_book_genre_not_from_books_genre_is_none change expected result
collector_lst change number of books
